### DISCUSSION TITLE
Getting $row information in the subfield of relationship.

### DISCUSSION THREAD

**Discussion by Uteev** [Laravel-Backpack/community-forum#703] [Q&A (Help)] [answered]
Hello to everyone! Here is my situation.

I have HasMany type of relationship. Basically **store** has many **zones**. So in the crud page of the store I use relationship field which handles zones. 

Zone in its turn has many **locations**. Location is a shelf in the store. And I need to show the user how many locations in each zone of the store. 

So in the relationship field for the zone I use 3 subfields: name, code, locations_qty. Name and code is easy as these are just text fields populated from db. And the last one locations_qty is a tricky one. I have created for it custom field and try to use it as a subfield. 

But to identify correct number of locations for the given zone I need to know the id of the zone. And I don't know how to get this information inside my custom subfield. 

I researched a little bit backpack's source code and identified that this information exists in $row variable in repeatable_row.blade.php. But all my attempts to access it failed and I keep getting exception "Undefined variable $row" . 

However if I make dd I can see this variable and its contents. But as soon as I try to access it throws before mentioned exception.

Can anyone help with this? 

**Basically I need to get the id of the current row relation in my custom subfield inside relationship field.**

**Comment by promatik** [comment]
Hi @Uteev, so each store has many zones, each zone has many locations, is that right?
You have a relationship in the store, to setup the zone, where you set the `locations_qty`.
How do you populate this field? Manually?

Shouldn't this value be computed based on the number of actual locations on the database?

**Reply by Uteev** [reply]
Hello @promatik, you are right. locations_qty should be just computed based on the number of actual locations. And actually I don't plan to have this field in db as I can always compute it. 

However in case of relationship field to compute the number of locations I should have access to the model of the zone. In that case I would just call $zone->loadCount('locations'). 

But as I understand it is not possible with Backpack relationship field. So I thought that if Backpack can't provide me with the zone model then maybe it can provide me at least with the id of the zone being processed. And after some research I actually find that info in a variable called $row in $repeatable_row.blade.php.

The problem is I can't access it and use its information. Or at least I can't find the way to access it.

So unfortunately I had to stop using the relationship field due to its limitation in this case and just create something similar that would work with my case. 

But I think that relationship field would be of much more greater use for developers if we can access data for each row. It would enable us to develop custom fields that can be used inside relationship field and can change its behavior based on the row. And the same works for the repeatable field. I think it is not very hard to implement.

**Reply by pxpm** [reply]
Hey @Uteev thanks for the report and the suggestion.
Talking specifically about the `$row` information, we don't pass the model further on, but maybe it will work if you create an `accessor` in your model ? Something like: 

```php
public function locationQty() :Attribute
{
return Attribute::make(
    get: fn ($value) => $value,
);
}
```
and add it in your model `$appends` property: `protected $appends = ['location_qty']` . 

Would that work for your scenario ?

Cheers

**Reply by Uteev** [reply, marked as answer]
Hi @pxpm , thanks for the suggestion!

Currently I just created the new custom field that shows all zones in rows and since it is my custom field I don't have the problem to have row information. Of course I had to do a bit of programming including javascript to replicate the behavior of backpack's relationship field but at the end I achieved what I needed. And by the way I totally removed locations_qty field from db as I compute this information dynamically.

Although I was able to achieve what I need still I think it would be a good use for many developers to have access to $row inside subfield.
